src/wallet.ts

```diff
@@ -124,3 +125,52 @@ export class Wallet extends AdapterL2(AdapterL1(ethers.Wallet)) {
         return await this.provider.broadcastTransaction(await this.signTransaction(populatedTx));
     }
 }
+
+interface SigningFunction {
+    (this: AbstractWallet, transaction: TransactionRequest): Promise<string>;
+}
+
+export class AbstractWallet extends Wallet {
+    readonly accountAddress: string;
+    paymaster: Paymaster;
+    customSigningFunction: SigningFunction;
+
+    constructor(
+        accountAddress: string,
+        privateKey: string,
+        providerL2?: Provider,
+        providerL1?: ethers.Provider,
+        paymaster: Paymaster | null = null,
+        customSigningFunction: SigningFunction = signingFunction,
+    ) {
+        super(privateKey, providerL2);
+        this.accountAddress = accountAddress;
+        this.paymaster = paymaster;
```

Comment:
why is `this.paymaster` here? How I can I control it's usage if it's here?

Reply:
`this.paymaster` is called in the `signingFunction` logic inside of `signTransaction` which is called for signing any transaction. The goal here was to provide a way for the developer to specify using a paymaster by default on any transaction. Currently the syntax in `signingFunction` is:

```ts
   ...
    transaction.customData = this.paymaster ? {
        gasPerPubdata: DEFAULT_GAS_PER_PUBDATA_LIMIT,
        paymasterParams: this.paymaster.getPaymasterParams(),
    } : {};
    ...
 ```
 
 But I think it would be better to change it actually to:
 
 ```ts
   ...
    transaction.customData ??= this.paymaster ? {
        gasPerPubdata: DEFAULT_GAS_PER_PUBDATA_LIMIT,
        paymasterParams: this.paymaster.getPaymasterParams(),
    } : {};
    ...
 ```
 
Considering the developer might want to specify some transactions on the frontend where a paymaster is applied, and some others not. But if the developer wants a paymaster to be used in all transactions, then they can simply declare the paymaster in this constructor here.